src/Phan/Analysis/ConditionVisitor.php

```diff
@@ -1074,13 +1075,18 @@ public function visitCall(Node $node): Context
             return $this->modifyComplexExpression($first_arg, $type_modification_callback, $this->context, $args);
         }
 
-        $function_name = \strtolower($raw_function_name);
-        if (\count($args) !== 1) {
-            if (!(\count($args) === 2 && \in_array($function_name, ['is_a', 'class_exists', 'method_exists'], true))) {
-                if (Config::getValue('redundant_condition_detection')) {
-                    $this->checkRedundantOrImpossibleTruthyCondition($node, $this->context, null, false);
+        // Check if a class-string is potentially being used as the first argument to is_a
+        if ($function_name === 'is_a' && isset($args[2])) {
+            $allow_string_type = UnionTypeVisitor::unionTypeFromNode($this->code_base, $this->context, $args[2])->containsTruthy();
+            if ($allow_string_type) {
+                // is_a("A"...) and is_a(A::class) early out in the node/var check above
+                $union_type = UnionTypeVisitor::unionTypeFromNode($this->code_base, $this->context, $first_arg);
+                if ($union_type->hasStringType()) {
```

Comment:
```suggestion
                // TODO: Narrow to class-string|object in another PR (low priority)
                if (!$union_type->isObject()) {
```

This should not fall through and should not convert mixed/empty to object